src/com/company/EchoServer.java

```diff
@@ -0,0 +1,51 @@
+package com.company;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class EchoServer {
+
+    private ServerSocket serverSocket;
+    private int portNumber;
+    private ServerMessenger serverMessenger;
+
+    public EchoServer(ServerSocket serverSocket, int portNumber, ServerMessenger serverMessenger) {
+        this.serverSocket = serverSocket;
+        this.portNumber = portNumber;
+        this.serverMessenger = serverMessenger;
+    }
+
+    public void run() {
+        serverMessenger.printServerPort(portNumber);
+        while (runServer()) {
+            connectWithSocket();
+        }
+    }
+
+    public boolean runServer() {
+        return true;
+    }
+
+    private void connectWithSocket() {
+        try {
+            Socket clientSocket = accept();
+            serverMessenger.informOfNewSocket();
+            echo(clientSocket);
```

Comment:
clientSocket/accept() could be inlined here

Reply:
done.